src/proof/PGTransformationAlgorithms.cc

```diff
@@ -555,73 +555,64 @@ void ProofGraph::proofTransformAndRestructure(const double left_time, const int
     }
 
     double init_time = cpuTime();
-    assert(!(max_num_loops > 0 && left_time > 0));
+    assert(not (max_num_loops > 0 and left_time > 0));
     //Flag to check if in a loop at least a transformation has been applied
     bool some_transf_done;
     long curr_num_loops = 0;
-    RuleContext ra1, ra2;
     std::deque<clauseid_t> q;
-    clauseid_t id;
     // Main external loop
     do {
```

Comment:
I think we can change this to `while (...)`

Reply:
Yes, I think we can, but we have to be careful about the loop iteration counter, since we increment it at the end of the loop.

Reply:
Changed. Nothing needs to change about the loop iteration counter, I was confused.